src/connector/src/sink/iceberg/mod.rs

```diff
@@ -519,6 +510,14 @@ impl Sink for IcebergSink {
 
     const SINK_NAME: &'static str = ICEBERG_SINK;
 
+    fn default_sink_decouple(desc: &SinkDesc) -> bool {
+        if let Some(interval) = desc.properties.get("commit_checkpoint_interval") {
+            interval.parse::<u64>().unwrap_or(0) > 1
+        } else {
+            false
+        }
+    }
```

Comment:
As mentioned by @wenym1, when users can `set sink_decouple = false`, sink will not be decoupled. If users use `commit_checkpoint_interval` at the same time, it could be problematic and cause a deadlock. Please verify this case.